python/lsst/analysis/tools/actions/plot/elements/plotInfoElement.py

```diff
@@ -0,0 +1,146 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("PlotInfoElement",)
+
+from typing import TYPE_CHECKING, Any, Mapping
+
+import matplotlib.transforms as transforms
+
+from ....interfaces import PlotElement
+
+if TYPE_CHECKING:
+    from matplotlib.axes import Axes
+
+
+class PlotInfoElement(PlotElement):
+    """Adds plotInfo to a figure.
+
+    Notes
+    ------
+    Converts a PlotElement dict to a string and creates
+    a textbox to display it.
+    """
+
+    def _buildPlotInfoText(self, plotInfo: Mapping[str, str]) -> str:
+        """Add useful information to the plot.
+
+        Parameters
+        ----------
+            A dictionary of the plot information.
+
+        Returns
+        -------
+        fig : `matplotlib.figure.Figure`
+            The figure with the information added.
+        """
+        # TO DO: This function is a near copy-paste of addPlotInfo
+        #        from plotUtils, but which only builds the plot info
+        #        text, rather than adding it to a figure. Suggest the
+        #        text-build is separated-out from addPlotInfo in plotUtils.
```

Comment:
This is an important TODO, and something we might want to consider now. Adding logic to parse a `plotInfo` dict here duplicates logic already inside `addPlotInfo` in `plotUtils`, and introduces the possibility of a divergence down the line.

Instead of duplicating code, it seems to me that the better approach would be (as you rightly point out in this TODO) to separate out the dict-parsing logic inside `addPlotInfo` and drop it into its own (recently freed-up name) `parsePlotInfo`. That should allow it to be imported both here an inside the remainder of `addPlotInfo` as appropriate. I don't think such an exercise is beyond the scope of this ticket - what do you think?